lib/audit_help.c

```diff
@@ -85,13 +84,14 @@ void audit_logger (int type, MAYBE_UNUSED const char *pgname, const char *op,
 		/*
 		 * The audit system needs white space in the op field to
 		 * be replaced with dashes so that parsers get the whole
-		 * field.
+		 * field. Not all C libraries have strdupa.
 		 */
-		char *fixed_op = strreplace (strdup (op), ' ', '-');
+		char *tmp_op = alloca (strlen (op) + 1);
+		strcpy (tmp_op, op);
+		char *fixed_op = strtr (tmp_op, ' ', '-');
```

Comment:
musl does have strdupa(3), AFAICS.

```sh

alx@devuan:~/src/musl/libc/master$ grepc strdupa .
./include/string.h:#define	strdupa(x)	strcpy(alloca(strlen(x)+1),x)
alx@devuan:~/src/musl/libc/master$ git blame -- include/string.h | grep strdupa
419ae6d5c (Rich Felker   2012-05-22 21:52:08 -0400  92) #define	strdupa(x)	strcpy(alloca(strlen(x)+1),x)
```

Reply:
Maybe it was strndupa? Anyways...ok.

Reply:
Yep, probably it was strndupa(3).  I had that issue recently.  <https://github.com/shadow-maint/shadow/pull/1189#issuecomment-2600900915>